api/v1alpha1/node_helpers.go

```diff
@@ -0,0 +1,51 @@
+package v1alpha1
+
+// validate the Node and NodeProperties passed are valid
+// e.g validate vmNames being unique, properties correctly defined, etc.
+// one day move to common/util
+
+import (
+	"fmt"
+)
+
+var (
+	typeOptions     = map[string]bool{"database": true, "haproxy": true}
+	roleOptions     = map[string]bool{"primary": true, "secondary": true}
+	failoverOptions = map[string]bool{"Automatic": true, "Manual": true}
+)
+
+func ValidateNodes(nodes []*Node, isHighAvailability bool) error {
+	if !isHighAvailability || len(nodes) < 5 {
+		return nil // no nodes is valid?
```

Comment:
None is valid right since we are defaulting?

Reply:
![image](https://github.com/irion4686/ndb-operator/assets/87503474/3002bdd3-a4a8-4d9d-a103-fd220daec3ba)

Looks like default is first